src/include/simeng/arch/aarch64/Architecture.hh

```diff
@@ -84,6 +81,10 @@ class Architecture : public arch::Architecture {
   /** Update the value of SVCRval_. */
   void setSVCRval(const uint64_t newVal) const;
```

Comment:
Currently, the Instruction class (at least for AArch64) holds a const reference to the Architecture. In exceptionHandler, we call the function `instruction.getArchitecture()` which returns said const reference. As such, we cannot make any changes to the underlying architecture object as it is held as a const reference. 

This should be changed, but I am unsure if it should happen in this PR, or in the planned re-working of the exceptionHandler logic... Additionally, what should the const reference held in `instruction` be changed to? Simply a reference?

Reply:
If the Instruction/Core need to make changes to the state in the `architecture` class then I agree that we'll need to change it to a reference. It was probably a const reference because we had no setter-style instructions. Given it goes beyond the scope of the `exceptionHandler`, I'd say it belongs in this PR.

Reply:
After trying to resolve this, it seems that this issue stems deeper than simply holding a const reference in the `Instruction` class. As discussed offline, I think fixing this / tidying up of the SVCR logic should be done in the re-factoring work that we have planned next